sw/device/lib/dif/dif_edn.h

```diff
@@ -134,6 +134,88 @@ typedef enum dif_edn_status {
   kDifEdnStatusCsrngAck,
 } dif_edn_status_t;
 
+/**
+ * EDN SM states.
+ */
+typedef enum dif_edn_sm_state {
+  /**
+   * Device is idle.
+   */
+  kDifEdnSmStateIdle = 389,
```

Comment:
Do the specific values of these enum entries mean something particular? If they do, we'll need a comment explaining them. If they don't, maybe it's better to leave the automatic values.

Also (more to the point!) I don't think this PR has any code that actually uses the new type. Can we remove this commit and add it to a different PR?

Reply:
Hi @rswarbrick,
Thanks for commenting. I'm using kDifEdnSmStateBootGenAckWait in this PR. The values are equivalent to the values of the EDN SM states. I'll make sure to add a comment to make this more clear.

Reply:
Duh, I grepped for the typename, not the values. D'oh!

And now I understand about the magic numbers. So you get things like
```
    CHECK_STATUS_OK(
        edn_testutils_wait_for_state(&edn1, kDifEdnSmStateBootGenAckWait));
```
and the point is that the hardware reports *some* magic number to Ibex and we avoid having to write any translation code by using that number as an enum value.

I guess it makes sense to keep e.g. `dif_edn_get_main_state_machine` using `uint32_t` output arguments because that way we aren't making claims about what the possible values are.

Cool, thanks for the explanation.